plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryPageSourceProvider.java

```diff
@@ -68,10 +70,12 @@ public ConnectorPageSource createPageSource(
         log.debug("createPageSource(transaction=%s, session=%s, split=%s, table=%s, columns=%s)", transaction, session, split, table, columns);
         BigQuerySplit bigQuerySplit = (BigQuerySplit) split;
 
-        // We expect columns list requested here to match list passed to ConnectorMetadata.applyProjection.
-        checkArgument(bigQuerySplit.getColumns().isEmpty() || bigQuerySplit.getColumns().equals(columns),
-                "Requested columns %s do not match list in split %s", columns, bigQuerySplit.getColumns());
-
+        Set<String> projectedColumnNames = bigQuerySplit.getColumns().stream().map(BigQueryColumnHandle::name).collect(Collectors.toSet());
+        // because we apply logic (download only parent columns - BigQueryMetadata.projectParentColumns)
+        // columns and split columns could differ
+        columns.stream()
+                .map(BigQueryColumnHandle.class::cast)
+                .forEach(column -> checkArgument(projectedColumnNames.contains(column.name()), "projected columns should contain all reader columns"));
```

Comment:
We could potentially create Set<String> for the parent column names in case of column handles provided by the split and passed as a part of PageSource -

Reply:
```
Set<String> projectedColumnNames = bigQuerySplit.getColumns().stream().map(BigQueryColumnHandle::name).collect(Collectors.toSet());
checkArgument(bigQuerySplit.getColumns().isEmpty() || bigQuerySplit.getColumns().map(BigQueryColumnHandle::name).collect(Collectors.toSet()).equals(columns),
                "Requested columns %s do not match list in split %s", columns, bigQuerySplit.getColumns());